diff --git a/lib/index.js b/lib/index.js
index 2dc42372aa661d1f6683dd6ae9ee71581c6dcde8..07c68b00dbd509150bddd68ee097a20f3fc38eca 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -72,7 +72,7 @@ class ReactRefreshPlugin {
       devServer: compiler.options.devServer,
       options: this.options,
     });
-    if (EntryPlugin) {
+    if (this.options.client !== false && EntryPlugin) {
       // Prepended entries does not care about injection order,
       // so we can utilise EntryPlugin for simpler logic.
       addEntries.prependEntries.forEach((entry) => {
@@ -139,7 +139,7 @@ class ReactRefreshPlugin {
           }
         );
       });
-    } else {
+    } else if (this.options.client !== false) {
       compiler.options.entry = injectRefreshEntry(compiler.options.entry, addEntries);
     }
 
diff --git a/lib/options.json b/lib/options.json
index 40443b8a8959fc691b43c712b18582f5ea48b399..9f62c9ea87b4b85035a94094059d43cda88b6fe9 100644
--- a/lib/options.json
+++ b/lib/options.json
@@ -55,6 +55,7 @@
     }
   },
   "properties": {
+    "client": { "type": "boolean" },
     "esModule": {
       "anyOf": [{ "type": "boolean" }, { "$ref": "#/definitions/ESModuleOptions" }]
     },
diff --git a/lib/utils/normalizeOptions.js b/lib/utils/normalizeOptions.js
index 8c7bbbb9fe14b16af67f64f2bd0b69efc4c5fea7..abf7ff04c3ac67ae6efdf508af4bdc3c04dde222 100644
--- a/lib/utils/normalizeOptions.js
+++ b/lib/utils/normalizeOptions.js
@@ -10,6 +10,7 @@ const normalizeOptions = (options) => {
   d(options, 'include', /\.([cm]js|[jt]sx?|flow)$/i);
   d(options, 'forceEnable');
   d(options, 'library');
+  d(options, 'client', true);
 
   n(options, 'overlay', (overlay) => {
     /** @type {import('../types').NormalizedErrorOverlayOptions} */
diff --git a/types/lib/types.d.ts b/types/lib/types.d.ts
index be0924c3a078ac910576c4952acb9862d5b20b1d..d176ae2f240095d7be9b0f59564b23271137fb06 100644
--- a/types/lib/types.d.ts
+++ b/types/lib/types.d.ts
@@ -39,6 +39,10 @@ export type NormalizedErrorOverlayOptions = import('type-fest').SetRequired<
   'entry' | 'module' | 'sockIntegration'
 >;
 export type ReactRefreshPluginOptions = {
+  /**
+   * Inject client modules.
+   */
+  client?: boolean | undefined;
   /**
    * Enables strict ES Modules compatible runtime.
    */
